cloudbridge/cloud/providers/aws/services.py

```diff
@@ -842,11 +842,46 @@ def instance_data(self):
         return [vm_type for vm_type in vm_types_list
                 if vm_type.get('pricing', {}).get(self.provider.region_name)]
 
+    @dispatch(event="provider.compute.vm_types.get",
+              priority=BaseVMTypeService.STANDARD_EVENT_PRIORITY)
+    def get(self, vm_type_id):
+        vm_type = (AWSVMType(self.provider, vm_type)
+                   for vm_type in self.instance_data
+                   if vm_type['instance_type'] == vm_type_id)
+        return next(vm_type, None)
+
+    @cachetools.cached(cachetools.TTLCache(maxsize=1, ttl=24 * 3600))
+    def _list_vm_types_in_zone(self, zone_name):
+        result = (self.provider.ec2_conn.meta.client
+                      .describe_reserved_instances_offerings(
+                          AvailabilityZone=zone_name))
```

Comment:
I feel we should filer these results a bit, namely, exclude `marketplace` instances, set `MinDuration` and `MaxDuration` to something much smaller than the defaults, and set `MaxInstanceCount` to 1 in this request. Maybe others as well. A subset of he filters can probably be applied in the next call lower. 

Docs: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeReservedInstancesOfferings.html